site/cypress/e2e/cypress-feature/late_days_cache.spec.js

```diff
@@ -0,0 +1,462 @@
+import {buildUrl} from '/cypress/support/utils.js';
+
+const predictedStatus = (days_allowed, days_late, remaining) => {
+
+    if (days_late === 0) {
+        return 'Good';
+    }
+    else if (days_late <= days_allowed && days_late <= remaining) {
+        // Valid
+        return 'Late';
+    }
+    else if (days_late > remaining) {
+        // Bad (too many for term)
+        return 'Bad (too many late days used this term)';
+    }
+    else {
+        // Bad (too many for assignment)
+        return 'Bad (too many late days used on this assignment)';
+    }
+};
```

Comment:
IS there a way to make this without branching ?

Reply:
In other files, this is the logic that is used to determine the status of a submission. I think we should keep it the same everywhere in the codebase.